File: algorithms/016.3Sum_Closest/3Sum_Closest.py
class Solution:
    def threeSumClosest(self, nums, target):
        nums.sort()
        i = 0 
        
        min_diff = 99
        result = 99
        
        while i < len(nums) - 2:
            j, k = i+1, len(nums)-1
            
            while j < k:
                diff = nums[i] + nums[j] + nums[k] - target
                if abs(diff) < min_diff:
                    min_diff = abs(diff)
                    result = nums[i] + nums[j] + nums[k]
                if diff < 0:
                    j = j + 1
                elif diff > 0:
                    k = k - 1
                else:
                    return result
            
            i = i + 1
            while  i < len(nums) - 2 and nums[i] == nums[i-1]:
                i = i+1
        return result

s = Solution()
a = s.threeSumClosest([1,1,-1,-1,3], -1)
print(a)

File: algorithms/016.3Sum_Closest/3Sum_Closest.java
<<<<<<< HEAD
public class Solution {
	public int threeSumClosest(int[] num, int target) {
		// Note: The Solution object is instantiated only once and is reused by
		// each test case.
		if (num == null || num.length < 3) {
			return Integer.MAX_VALUE;
		}
		Arrays.sort(num);
		int closet = Integer.MAX_VALUE / 2; // otherwise it will overflow for opeartion (closet-target)'
		for (int i = 0; i < num.length - 2; i++) {
			int left = i + 1;
			int right = num.length - 1;
			while (left < right) {
				int sum = num[i] + num[left] + num[right];
				if (sum == target) {
					return sum;
				} else if (sum < target) {
					left++;
				} else {
					right--;
				}
				closet = Math.abs(sum - target) < Math.abs(closet - target) ? sum : closet;
			}
		}
		return closet;
	}
=======
public class Solution {
	public int threeSumClosest(int[] num, int target) {
		// Note: The Solution object is instantiated only once and is reused by
		// each test case.
		if (num == null || num.length < 3) {
			return Integer.MAX_VALUE;
		}
		Arrays.sort(num);
		int closet = Integer.MAX_VALUE / 2; // otherwise it will overflow for opeartion (closet-target)'
		for (int i = 0; i < num.length - 2; i++) {
			int left = i + 1;
			int right = num.length - 1;
			while (left < right) {
				int sum = num[i] + num[left] + num[right];
				if (sum == target) {
					return sum;
				} else if (sum < target) {
					left++;
				} else {
					right--;
				}
				closet = Math.abs(sum - target) < Math.abs(closet - target) ? sum : closet;
			}
		}
		return closet;
	}
>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74
}